### PR TITLE
chore: localize skills/commands and add cross-agent config

### LINKED ISSUE #636: chore: localize shared skills/commands into project and ensure cross-agent compatibility
## Summary

Currently, some Claude Code skills and commands are sourced from the marketplace (shared/generic). These don't always align with the ry project's specific workflow (AGENTS.md, branching rules, TDD process, etc.), which reduces rule adherence. This issue covers:

1. Localizing shared skills/commands into the project (`.claude/`) and customizing them for ry's workflow
2. Ensuring consistent development experience across AI coding agents (Claude Code, Codex, Cursor)

### Problem

- Shared marketplace skills are generic — they don't know about ry's branching rules (`vx.x.x` branches), TDD workflow, issue-driven development, or AGENTS.md conventions
- When rules in shared skills conflict with project-specific workflow, the agent's compliance degrades
- Example: `.claude/skills/run-ry/SKILL.md` is already localized — other skills/commands should follow

### Proposed approach

- Copy relevant marketplace skills into `.claude/skills/` and `.claude/commands/`
- Customize each to align with AGENTS.md workflow:
  - Git operations: respect branching rules, commit conventions, PR workflow
  - Test execution: use project-specific build/test commands
  - Code review: understand ry's codegen architecture, stdlib patterns
- Remove marketplace dependency for customized skills so updates don't overwrite local changes

### Problem

The development workflow defined in AGENTS.md should produce consistent results regardless of which AI coding agent is used. Currently:

- **Claude Code**: Full support via CLAUDE.md + AGENTS.md + skills
- **Codex**: Can read AGENTS.md but lacks skill/command infrastructure
- **Cursor**: Has its own rules system (.cursorrules) but doesn't read CLAUDE.md/AGENTS.md

### Proposed approach

- Identify which workflow rules are agent-agnostic (can be expressed in a shared format) vs agent-specific
- Create agent-specific configuration files that reference the shared workflow:
  - `.cursorrules` or `.cursor/rules` — Cursor-compatible rules derived from AGENTS.md
  - `codex.md` or equivalent — Codex-compatible instructions
- Keep AGENTS.md as the single source of truth — agent-specific files should reference or derive from it, not duplicate
- Document which capabilities are agent-specific and cannot be replicated (e.g., Claude Code skills, MCP tools)

### Limitations

Some capabilities are Claude Code specific and cannot be replicated:
- Skills/commands infrastructure
- MCP server integration (GitHub MCP, etc.)
- Hook system

For these, document the expected manual workflow for other agents.

## Acceptance criteria

- [ ] All actively used marketplace skills/commands localized into `.claude/`
- [ ] Localized skills customized to align with AGENTS.md workflow
- [ ] Cursor-compatible rules file created and tested
- [ ] Codex-compatible instructions created and tested
- [ ] AGENTS.md remains the single source of truth
- [ ] Documentation of agent-specific limitations and manual workarounds

### PR DESCRIPTION
**Pull request by t0k0sh1**
## Summary

Closes #636

- Replace `.claude/commands/` with `.claude/skills/` — 14 skills with `git-` prefix, flat structure
- Mirror all 14 skills to `.codex/skills/` for Codex CLI compatibility
- Rename `run-ry` → `ry-playground`, `branch-naming` → `git-branch-naming`, `triage-issue` → `git-triage-issue`
- Customize skills for ry workflow: branch safety checks, open PR (not draft), auto commit+push after review fixes, non-default branch merge warnings
- Move build & ASan sections from `CLAUDE.md` to `AGENTS.md` (single source of truth)
- Simplify `CLAUDE.md` to `@AGENTS.md` reference only
- Add `.cursorrules` for Cursor AI compatibility (derived from AGENTS.md)
- Track `.mcp.json` (serena MCP server config)

## Test plan

- [ ] Verify `/git-commit` refuses on `main` and `vx.x.x` branches
- [ ] Verify `/git-commit-push-pr` creates open PR (not draft)
- [ ] Verify `/git-fix-pr-reviews` auto-pushes after applying fixes
- [ ] Verify `.codex/skills/` mirrors `.claude/skills/` (14 skills each)
- [ ] Verify `.cursorrules` contains build commands and branch rules

## Summary by CodeRabbit

* **New Features**
  * Added many Git workflow automations (branch naming, commit, push, PR open/merge, push+PR, switch/sync, resolve conflicts, search/triage issues, fix PR reviews) and an interactive Ry playground for ad‑hoc runs.

* **Documentation**
  * Expanded developer guidance and consolidated build/test and ASan workflows into central documentation; updated agent/skill usage guidance.

* **Chores**
  * Added local agent/configuration manifest and reorganized skill docs (several legacy docs removed, many new ones added).

### CODE REVIEW

File: .claude/skills/git-triage-issue/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+description: When a problem is found outside the current task scope, ask the user how to handle it. Aligns with AGENTS.md scope-out rules.
+allowed-tools: Bash(gh issue create:*), Bash(gh search issues:*), Bash(gh issue view:*), Bash(gh issue comment:*), mcp__plugin_github_github__search_issues
+metadata:
+  short-description: Triage out-of-scope issues
+---
+
+# Git Triage Issue
+
+When you encounter a problem, bug, review comment, or improvement opportunity that may be outside the current task scope, **always** ask the user how to handle it. Never silently skip or auto-decide.
+
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-triage-issue/SKILL.md]: Fixed: the intro text has been updated to align with the Decision Logic (autonomous for Case 1/2, ask only for Case 3).

File: .codex/skills/git-triage-issue/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+name: git-triage-issue
+description: When a problem is found outside the current task scope, ask the user how to handle it. Aligns with AGENTS.md scope-out rules.
+metadata:
+  short-description: Triage out-of-scope issues
+---
+
+# Git Triage Issue
+
+When you encounter a problem, bug, review comment, or improvement opportunity that may be outside the current task scope, **always** ask the user how to handle it. Never silently skip or auto-decide.
+
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-triage-issue/SKILL.md]: Fixed: the intro text has been updated to align with the Decision Logic (autonomous for Case 1/2, ask only for Case 3).

File: .claude/skills/git-fix-pr-reviews/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+- Otherwise, use the PR associated with the current branch (from the Context above)
+- If no PR is found, display the following and stop:
+  > No PR found. Run this command on a branch with an associated PR, or specify a PR number.
+
+### Step 2: Fetch review comments
+
+Get repository info with `gh repo view --json owner,name --jq '.owner.login + "/" + .name'` and call the following two APIs **in parallel**:
+
+1. `gh api repos/{owner}/{repo}/pulls/{number}/comments` — inline comments (attached to specific files/lines)
+2. `gh api repos/{owner}/{repo}/pulls/{number}/reviews` — review summaries (general comments)
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-fix-pr-reviews/SKILL.md]: Fixed: added `--paginate` to both `gh api` calls.

File: .codex/skills/git-fix-pr-reviews/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+- Otherwise, use the PR associated with the current branch (from the Context above)
+- If no PR is found, display the following and stop:
+  > No PR found. Run this command on a branch with an associated PR, or specify a PR number.
+
+### Step 2: Fetch review comments
+
+Get repository info with `gh repo view --json owner,name --jq '.owner.login + "/" + .name'` and call the following two APIs **in parallel**:
+
+1. `gh api repos/{owner}/{repo}/pulls/{number}/comments` — inline comments (attached to specific files/lines)
+2. `gh api repos/{owner}/{repo}/pulls/{number}/reviews` — review summaries (general comments)
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-fix-pr-reviews/SKILL.md]: Fixed: added `--paginate` to both `gh api` calls.

File: .cursorrules [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+6. Tests — import tests + native function execution tests
+
+## Key Conventions
+
+### Git
+
+- **Branch naming**: `<type>/<short-description>` (feat, fix, docs, refactor, test, chore)
+- **Never commit on `main` or `vx.x.x` branches** — always create a feature branch
+- **PR base**: the branch you forked from (often `vx.x.x`, not always `main`)
+- **Commit messages**: conventional commits (feat:, fix:, docs:, refactor:, test:, chore:)
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .cursorrules:55]: Fixed: unified to `v*.*.*` glob pattern throughout.

File: .claude/skills/git-branch-naming/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+- `fix/utf8-overread` — runtime bug fix
+- `docs/update-reference-types` — documentation update
+- `refactor/parser-cleanup` — internal restructuring
+- `chore/pre-release-v0.0.8` — release preparation
+- `test/add-list-edge-cases` — test coverage improvement
+
+## Repository Rules
+
+- Always branch from the current release branch (`vx.x.x`) or `main`
+- Never commit directly on `main` or `vx.x.x` branches
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-branch-naming/SKILL.md]: Fixed: unified to `v*.*.*` glob pattern throughout.

File: .codex/skills/git-branch-naming/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+- `fix/utf8-overread` — runtime bug fix
+- `docs/update-reference-types` — documentation update
+- `refactor/parser-cleanup` — internal restructuring
+- `chore/pre-release-v0.0.8` — release preparation
+- `test/add-list-edge-cases` — test coverage improvement
+
+## Repository Rules
+
+- Always branch from the current release branch (`vx.x.x`) or `main`
+- Never commit directly on `main` or `vx.x.x` branches
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-branch-naming/SKILL.md]: Fixed: unified to `v*.*.*` glob pattern throughout.

File: .claude/skills/git-commit-push-pr/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+- Current git diff (staged and unstaged changes): !`git diff HEAD`
+- Current branch: !`git branch --show-current`
+
+## Steps
+
+### 1. Create branch (if needed)
+
+If the current branch is `main` or matches `v*.*.*` (e.g. `v0.0.8`), you **MUST** create a new feature branch before committing.
+- Use the `<type>/<short-description>` naming convention
+- Never commit directly to `main` or `vx.x.x` branches
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-commit-push-pr/SKILL.md]: Fixed: unified to `v*.*.*` glob pattern throughout.

File: .codex/skills/git-commit-push-pr/SKILL.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+- Current git diff (staged and unstaged changes): !`git diff HEAD`
+- Current branch: !`git branch --show-current`
+
+## Steps
+
+### 1. Create branch (if needed)
+
+If the current branch is `main` or matches `v*.*.*` (e.g. `v0.0.8`), you **MUST** create a new feature branch before committing.
+- Use the `<type>/<short-description>` naming convention
+- Never commit directly to `main` or `vx.x.x` branches
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-commit-push-pr/SKILL.md]: Fixed: unified to `v*.*.*` glob pattern throughout.

File: .codex/skills/git-branch-naming/SKILL.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+- The feature branch will be PR'd back to its parent branch
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-branch-naming/SKILL.md]: Fixed: unified to `v*.*.*` glob pattern throughout.

File: .claude/skills/git-fix-pr-reviews/SKILL.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+
+**Gate**: Check the PR for unresolved review threads using the GraphQL API:
+
+```bash
+gh api graphql --paginate -f query='
+query($owner:String!,$repo:String!,$pr:Int!,$endCursor:String) {
+  repository(owner:$owner,name:$repo) {
+    pullRequest(number:$pr) {
+      reviewThreads(first:100,after:$endCursor) {
+        nodes { isResolved }
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-fix-pr-reviews/SKILL.md]: Moot: the auto commit+push gate (Step 7) has been removed. The skill no longer commits or pushes.

File: .codex/skills/git-fix-pr-reviews/SKILL.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+
+**Gate**: Check the PR for unresolved review threads using the GraphQL API:
+
+```bash
+gh api graphql --paginate -f query='
+query($owner:String!,$repo:String!,$pr:Int!,$endCursor:String) {
+  repository(owner:$owner,name:$repo) {
+    pullRequest(number:$pr) {
+      reviewThreads(first:100,after:$endCursor) {
+        nodes { isResolved }
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-fix-pr-reviews/SKILL.md]: Moot: the auto commit+push gate (Step 7) has been removed. The skill no longer commits or pushes.

File: .claude/skills/git-fix-pr-reviews/SKILL.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+}' -f owner='{owner}' -f repo='{repo}' -F pr='{number}' \
+  --jq '[.data.repository.pullRequest.reviewThreads.nodes[] | select(.isResolved | not)] | length' \
+  | awk '{s+=$1} END {print s+0}'
+```
+
+- If the count is **greater than 0**, display the count and stop:
+  > <N> unresolved review thread(s) remain. Commit and push skipped — resolve all threads first.
+- If **0**, proceed with commit and push.
+
+Before committing, verify the current branch is not `main` or `v*.*.*`. If it is, stop and report the issue.
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-fix-pr-reviews/SKILL.md]: Moot: the auto commit+push gate (Step 7) has been removed. The skill no longer commits or pushes.

File: .claude/skills/git-commit-push-pr/SKILL.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-commit-push-pr/SKILL.md]: Intentional: branch creation is conditional — only required when on `main` or `v*.*.*`. If already on a feature branch, creating another one is wrong. The skill correctly handles both cases.

File: .codex/skills/git-commit-push-pr/SKILL.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-commit-push-pr/SKILL.md:25]: Intentional: branch creation is conditional — only required when on `main` or `v*.*.*`. If already on a feature branch, creating another one is wrong. The skill correctly handles both cases.

File: .claude/skills/git-sync-branch/SKILL.md [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+- Available remote branches: !`git branch -r`
+
+## Inputs
+
+User input: $ARGUMENTS
+
+## Steps
+
+1. Extract the target branch name from the user's input. Default to `main` if not specified.
+2. **Safety check**: If the current branch is the same as the target branch, report "Already on the target branch — no sync needed" and stop.
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-sync-branch/SKILL.md:27]: Intentional: this skill is for cross-branch merges (e.g. merge `main` into feature branch). Same-branch remote sync should use `git pull` directly, not this skill.

File: .codex/skills/git-sync-branch/SKILL.md [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+## Inputs
+
+User input: $ARGUMENTS
+
+## Steps
+
+1. Extract the target branch name from the user's input. Default to `main` if not specified.
+2. **Safety check**: If the current branch is the same as the target branch, report "Already on the target branch — no sync needed" and stop.
+3. Run `git fetch origin` to get the latest remote state.
+4. Run `git merge origin/<target>` to merge the target branch changes into the current branch.
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-sync-branch/SKILL.md:28]: Intentional: the default `main` is appropriate — users explicitly specify the target branch when syncing. The skill description clearly states the direction: `origin/<target> → current branch`.

File: .claude/skills/git-open-pr/SKILL.md [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+## Inputs
+
+User input: $ARGUMENTS
+
+## Steps
+
+### Step 1: Identify PR
+
+- If the user specified a PR number (e.g. `#123` or `123`), use that number
+- Otherwise, use the PR associated with the current branch (from the Context above)
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-open-pr/SKILL.md:27]: Not applicable: `gh pr view` and `gh pr ready` handle `#123` input correctly — the `#` prefix is stripped by the CLI automatically.

File: .codex/skills/git-fix-pr-reviews/SKILL.md [hunk 18]
```diff
@@ -0,0 +1,80 @@
+---
+name: git-fix-pr-reviews
+description: Fetch PR review comments, triage them, apply fixes, and automatically commit and push the changes.
+metadata:
+  short-description: Triage and fix PR review comments
+---
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-fix-pr-reviews/SKILL.md:6]: By design: Codex CLI does not support `allowed-tools` in SKILL.md frontmatter. Only `.claude/skills/` uses this field. The `.codex/` versions intentionally omit it.

File: .codex/skills/git-open-pr/SKILL.md [hunk 19]
```diff
@@ -0,0 +1,44 @@
+---
+name: git-open-pr
+description: Mark a draft pull request as ready for review.
+metadata:
+  short-description: Mark a draft PR as ready for review
+---
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-open-pr/SKILL.md:6]: By design: Codex CLI does not support `allowed-tools` in SKILL.md frontmatter. Only `.claude/skills/` uses this field. The `.codex/` versions intentionally omit it.

File: .codex/skills/git-triage-issue/SKILL.md [hunk 20]
```diff
@@ -0,0 +1,44 @@
+---
+name: git-triage-issue
+description: When a problem is found outside the current task scope, ask the user how to handle it. Aligns with AGENTS.md scope-out rules.
+metadata:
+  short-description: Triage out-of-scope issues
+---
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-triage-issue/SKILL.md:6]: By design: Codex CLI does not support `allowed-tools` in SKILL.md frontmatter. Only `.claude/skills/` uses this field. The `.codex/` versions intentionally omit it.

File: .claude/skills/git-push/SKILL.md [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+- Recent commits: !`git log --oneline -10`
+- Unpushed commits: !`git log --oneline @{u}..HEAD 2>/dev/null || echo "No upstream set"`
+
+## Steps
+
+### 1. Merge base branch
+
+1. Detect the base branch:
+   - Run `gh pr view --json baseRefName -q .baseRefName` to get the base branch of the PR
+   - If no PR exists, detect the parent branch:
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-push/SKILL.md:25]: Works as designed: `gh pr view` is tried first and if it fails (no PR exists), the fallback logic detects the parent `v*.*.*` branch or defaults to `main`. The error case is explicitly handled in the instructions.

File: .codex/skills/git-push/SKILL.md [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+### 1. Merge base branch
+
+1. Detect the base branch:
+   - Run `gh pr view --json baseRefName -q .baseRefName` to get the base branch of the PR
+   - If no PR exists, detect the parent branch:
+     - Check if a `v*.*.*` branch exists locally that is an ancestor of the current branch
+     - If found, use that as the base branch
+     - Otherwise default to `main`
+2. Run `git fetch origin` to get the latest remote state
+3. Run `git merge origin/<base branch>` to merge the base branch changes
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-push/SKILL.md:29]: Works as designed: `gh pr view` is tried first and if it fails (no PR exists), the fallback logic detects the parent `v*.*.*` branch or defaults to `main`. The error case is explicitly handled in the instructions.

File: .claude/skills/git-commit-push/SKILL.md [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+
+### 1. Commit
+
+Create a single commit with an appropriate message based on the changes. Use conventional commit format.
+
+### 2. Merge base branch
+
+1. Detect the base branch:
+   - Run `gh pr view --json baseRefName -q .baseRefName` to get the base branch of the PR
+   - If no PR exists, detect the parent branch:
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-commit-push/SKILL.md:35]: Works as designed: `gh pr view` is tried first and if it fails (no PR exists), the fallback logic detects the parent `v*.*.*` branch or defaults to `main`. The error case is explicitly handled in the instructions.

File: .claude/skills/git-commit-push-pr/SKILL.md [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+
+### 2. Commit
+
+Create a single commit with an appropriate message. Use conventional commit format.
+
+### 3. Merge base branch
+
+1. Detect the base branch:
+   - Run `gh pr view --json baseRefName -q .baseRefName` to get the base branch of the PR
+   - If no PR exists, detect the parent branch:
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-commit-push-pr/SKILL.md:35]: Works as designed: `gh pr view` is tried first and if it fails (no PR exists), the fallback logic detects the parent `v*.*.*` branch or defaults to `main`. The error case is explicitly handled in the instructions.

File: .codex/skills/git-commit-push-pr/SKILL.md [hunk 25]
```diff
@@ -0,0 +1,57 @@
...
+### 2. Commit
+
+Create a single commit with an appropriate message. Use conventional commit format.
+
+### 3. Merge base branch
+
+1. Detect the base branch:
+   - Run `gh pr view --json baseRefName -q .baseRefName` to get the base branch of the PR
+   - If no PR exists, detect the parent branch:
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-commit-push-pr/SKILL.md:34]: Works as designed: `gh pr view` is tried first and if it fails (no PR exists), the fallback logic detects the parent `v*.*.*` branch or defaults to `main`. The error case is explicitly handled in the instructions.

File: .codex/skills/git-commit-push/SKILL.md [hunk 26]
```diff
@@ -0,0 +1,50 @@
...
+
+### 1. Commit
+
+Create a single commit with an appropriate message based on the changes. Use conventional commit format.
+
+### 2. Merge base branch
+
+1. Detect the base branch:
+   - Run `gh pr view --json baseRefName -q .baseRefName` to get the base branch of the PR
+   - If no PR exists, detect the parent branch:
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-commit-push/SKILL.md:34]: Works as designed: `gh pr view` is tried first and if it fails (no PR exists), the fallback logic detects the parent `v*.*.*` branch or defaults to `main`. The error case is explicitly handled in the instructions.

File: .mcp.json [hunk 27]
```diff
@@ -0,0 +1,16 @@
+{
+  "mcpServers": {
+    "serena": {
+      "command": "uvx",
+      "args": [
+        "--from",
+        "git+https://github.com/oraios/serena",
+        "serena-mcp-server",
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .mcp.json:8]: Acknowledged: we intentionally use the latest version of serena for now. Pinning to a specific commit would reduce flexibility for a dev-only tool. Will revisit if reproducibility becomes a concern.

[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .mcp.json:8]: Acknowledged: we intentionally use the latest version of serena for now. Pinning to a specific commit would reduce flexibility for a dev-only tool. Will revisit if reproducibility becomes a concern.

File: .mcp.json [hunk 28]
```diff
@@ -0,0 +1,16 @@
...
+        "--context",
+        "ide-assistant",
+        "--enable-web-dashboard",
+        "false"
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .mcp.json:12]: Acknowledged: we intentionally use the latest version of serena for now. Pinning to a specific commit would reduce flexibility for a dev-only tool. Will revisit if reproducibility becomes a concern.

File: .claude/skills/git-sync-branch/SKILL.md [hunk 29]
```diff
@@ -0,0 +1,31 @@
+---
+name: git-sync-branch
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-sync-branch/SKILL.md:2]: Acknowledged: the name was explicitly chosen by the project owner. `git-sync-branch` is clear enough in context — it syncs a target branch into the current one.

File: .claude/skills/git-sync-branch/SKILL.md [hunk 30]
```diff
@@ -0,0 +1,31 @@
...
+User input: $ARGUMENTS
+
+## Steps
+
+1. Extract the target branch name from the user's input. Default to `main` if not specified.
+2. **Safety check**: If the current branch is the same as the target branch, report "Already on the target branch — no sync needed" and stop.
+3. Run `git fetch origin` to get the latest remote state.
+4. Run `git merge origin/<target>` to merge the target branch changes into the current branch.
+5. **On conflict**: Report the conflict to the user and let them decide how to proceed. Do NOT auto-abort the merge.
+6. **On success**: Report the merge result and a summary including the number of commits merged.
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-sync-branch/SKILL.md:35]: Acknowledged: adding a dirty-worktree check would add complexity. The agent executing this skill can handle merge failures gracefully.

File: .codex/skills/git-search-issues/SKILL.md [hunk 31]
```diff
@@ -0,0 +1,105 @@
...
+- **Do not use `state:open` or `state:closed`** in `search_issues` queries (returns 0 results). Use `is:open` / `is:closed` instead.
+
+## Input
+
+User input: `$ARGUMENTS`
+
+Behavior depends on the input:
+
+### No arguments, or "find next issue", "search", etc.
+
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-search-issues/SKILL.md:29]: Acknowledged: the routing logic is clear from context — exact match on known keywords triggers default flow, anything else is a keyword search.

File: .codex/skills/git-search-issues/SKILL.md [hunk 32]
```diff
@@ -0,0 +1,105 @@
...
+- state: `"OPEN"`
+- labels: `["specified label"]`
+- orderBy: `"CREATED_AT"`, direction: `"DESC"`
+- perPage: `20`
+
+### Keyword search (e.g. `HTTP`, `lambda`)
+
+Call `search_issues`:
+- query: `"is:open keyword"`
+- owner: `"t0k0sh1"`, repo: `"ry"`
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-search-issues/SKILL.md:63]: Acknowledged: the `<keyword>` placeholder was already clarified in a previous fix. Multi-word escaping is handled by the MCP tool layer.

File: .codex/skills/git-triage-issue/SKILL.md [hunk 33]
```diff
@@ -0,0 +1,44 @@
...
+
+### Case 2: Future improvement (not caused by the current change)
+
+If the issue is a future improvement unrelated to the current change:
+1. Search existing issues with `gh search issues` to avoid duplicates
+2. If a duplicate exists, add a comment to the existing issue with relevant context
+3. Otherwise, create a new issue with `gh issue create`:
+   - A clear, descriptive title
+   - A body that includes the context (what, where, how it was found)
+4. Report the issue number and title to the user
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-triage-issue/SKILL.md:37]: Acknowledged: error handling for `gh` commands is standard CLI behavior — the agent will report failures naturally. Adding explicit guidance would over-specify the skill.

File: .claude/skills/git-branch-naming/SKILL.md [hunk 34]
```diff
@@ -0,0 +1,55 @@
...
+Create a new git branch following the `<type>/<short-description>` naming convention.
+
+## Branch Name Format
+
+```text
+<type>/<short-description>
+```
+
+- **type**: one of the Conventional Commits types listed below
+- **short-description**: concise English summary in kebab-case (2-4 words ideal)
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-branch-naming/SKILL.md:20]: Acknowledged: "2-4 words ideal" is a guideline, not a hard rule. The examples make the intent clear.

File: .claude/skills/git-branch-naming/SKILL.md [hunk 35]
```diff
@@ -0,0 +1,55 @@
...
+
+| Type | When to use |
+|------|-------------|
+| `feat` | New feature or functionality |
+| `fix` | Bug fix |
+| `docs` | Documentation only |
+| `style` | Formatting, whitespace, no code change |
+| `refactor` | Code restructuring without behavior change |
+| `test` | Adding or updating tests |
+| `chore` | Build, CI, dependencies, tooling |
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-branch-naming/SKILL.md:32]: Not applicable: `git checkout -b` is correct and works as intended for branch creation.

File: .claude/skills/git-branch-naming/SKILL.md [hunk 36]
```diff
@@ -0,0 +1,55 @@
...
+- `docs/update-reference-types` — documentation update
+- `refactor/parser-cleanup` — internal restructuring
+- `chore/pre-release-v0.0.8` — release preparation
+- `test/add-list-edge-cases` — test coverage improvement
+
+## Repository Rules
+
+- Always branch from the current release branch (`vx.x.x`) or `main`
+- Never commit directly on `main` or `vx.x.x` branches
+- The feature branch will be PR'd back to its parent branch
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-branch-naming/SKILL.md:47]: Acknowledged: when already on a feature branch, the skill is not invoked — users call this when starting new work.

File: .claude/skills/git-sync-branch/SKILL.md [hunk 37]
```diff
@@ -0,0 +1,35 @@
...
+## Steps
+
+1. Extract the target branch name from the user's input. Default to `main` if not specified.
+2. **Safety check**: If the current branch is the same as the target branch, report "Already on the target branch — no sync needed" and stop.
+3. Run `git fetch origin` to get the latest remote state.
+4. Run `git merge origin/<target>` to merge the target branch changes into the current branch.
+5. **On conflict**: Report the conflict to the user with the list of conflicting files and suggest available options:
+   - `git status` to inspect conflicts
+   - Edit files to resolve, then `git add <file>` and `git commit` to complete
+   - `git merge --abort` to cancel the merge
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-sync-branch/SKILL.md:33]: Acknowledged: using `git commit` for merge completion is correct — it's the standard way to finalize a merge after resolving conflicts. Other skills use `git merge --continue` which is equivalent.

File: .claude/skills/git-triage-issue/SKILL.md [hunk 38]
```diff
@@ -0,0 +1,45 @@
...
+- A fix requires changes that extend beyond the original scope
+- Any situation where you are unsure whether something should be addressed now
+
+## Decision Logic
+
+Classify the issue and act **autonomously** (do not ask the user for permission):
+
+### Case 1: Caused by the current change
+
+If the issue is caused by the current change (including indirect causes), or if it did not exist prior to this change:
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-triage-issue/SKILL.md:26]: Acknowledged: "prior to" is fine here — not worth a change for pure style preference.

File: .codex/skills/git-switch-branch/SKILL.md [hunk 39]
```diff
@@ -0,0 +1,30 @@
...
+## Inputs
+
+User input: $ARGUMENTS
+
+## Steps
+
+1. Extract the branch name from the user's input. If no branch is specified, default to `main`.
+2. Run `git fetch origin` to refresh remote branch information.
+3. Check the available branches to confirm the branch exists (local or remote).
+4. If the branch does not exist locally or remotely, notify the user and stop.
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-switch-branch/SKILL.md:26]: Acknowledged: the phrasing is clear enough in context.

File: .codex/skills/git-switch-branch/SKILL.md [hunk 40]
```diff
@@ -0,0 +1,30 @@
...
+5. If the branch exists locally, run `git switch <branch>`.
+6. If the branch exists only on remote, run `git switch --track origin/<branch>`.
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-switch-branch/SKILL.md:28]: Acknowledged: sentence structure variation is a style preference, not a clarity issue.

File: .codex/skills/git-sync-branch/SKILL.md [hunk 41]
```diff
@@ -0,0 +1,34 @@
...
+5. **On conflict**: Report the conflict to the user with the list of conflicting files and suggest available options:
+   - `git status` to inspect conflicts
+   - Edit files to resolve, then `git add <file>` and `git commit` to complete
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .codex/skills/git-sync-branch/SKILL.md:31]: Acknowledged: same as the .claude counterpart — `git commit` is correct for completing a merge.

File: .claude/skills/git-commit-push/SKILL.md [hunk 42]
```diff
@@ -0,0 +1,51 @@
...
+   - Run `git diff --name-only --diff-filter=U` to list conflicting files
+   - `Read` each conflicting file and examine the conflict markers
+   - Use `Edit` to resolve the conflicts
+   - Stage each resolved file with `git add`
+   - Once all conflicts are resolved, run `git merge --continue`
+   - If any conflict cannot be resolved, report to the user and stop (do NOT auto-abort the merge)
+
+### 3. Push
+
+Push the branch to origin.
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply .claude/skills/git-commit-push/SKILL.md:51]: Not applicable: `git-commit-push` is for branches that already have upstream tracking set. New branch creation (with `-u`) is handled by `git-commit-push-pr`. Adding `-u` here would be redundant for the intended use case.
